bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerCreateOp.java

```diff
@@ -261,15 +269,36 @@ public CreateBuilder withDigestType(org.apache.bookkeeper.client.api.DigestType
         }
 
         @Override
+        public CreateBuilder withLedgerType(LedgerType ledgerType) {
+            this.builderLedgerType = ledgerType;
+            return this;
+        }
+
+        @Override
         public CreateAdvBuilder makeAdv() {
             return new CreateAdvBuilderImpl(this);
         }
 
         private boolean validate() {
+            if (builderLedgerType == null) {
+                LOG.error("invalid null builderLedgerType");
+                return false;
+            }
+
+            if (bk.getConf().getUseV2WireProtocol() && builderLedgerType != LedgerType.FORCE_ON_JOURNAL) {
+                LOG.error("invalid builderLedgerType {} while using v2 wire protocol");
+                return false;
+            }
+
             if (builderWriteQuorumSize > builderEnsembleSize) {
                 LOG.error("invalid writeQuorumSize {} > ensembleSize {}", builderWriteQuorumSize, builderEnsembleSize);
                 return false;
             }
+            if (builderLedgerType.equals(LedgerType.FORCE_DEFERRED_ON_JOURNAL) && builderWriteQuorumSize != builderEnsembleSize) {
```

Comment:
line 288 is using `!=` and line 297 is using `equals`. can we at least using same pattern?

Reply:
I will use equals everywhere. Fixing

Reply:
@sijie  I think I have addressed this one, the link is broken with a rebase